pkg/virt-manifest/rest/manifest.go

```diff
@@ -0,0 +1,69 @@
+package rest
+
+import (
+	"github.com/emicklei/go-restful"
+	"golang.org/x/net/context"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/middleware"
+	"kubevirt.io/kubevirt/pkg/rest"
+	"kubevirt.io/kubevirt/pkg/rest/endpoints"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+	"kubevirt.io/kubevirt/pkg/virt-manifest"
+)
+
+type manifest struct {
+	libvirtUri    string
+	libvirtUser   string
+	libvirtPasswd string
+}
+
+func ManifestService(libvirtUri string, libvirtUser string, libvirtPasswd string) (*restful.WebService, error) {
+	ws := new(restful.WebService)
+
+	m := manifest{
+		libvirtUri:    libvirtUri,
+		libvirtUser:   libvirtUser,
+		libvirtPasswd: libvirtPasswd,
+	}
+
+	ctx := context.Background()
+	target := endpoints.MakeGoRestfulWrapper(endpoints.NewHandlerBuilder().Post(&v1.VM{}).Endpoint(m.mapManifest).Build(ctx))
+
+	ws.Path("/").
+		Consumes(restful.MIME_JSON, rest.MIME_YAML).
+		Produces(restful.MIME_JSON, rest.MIME_YAML)
+	ws.Route(ws.GET("/api/v1/status").To(alive))
+	ws.Route(ws.POST("/apis/" + v1.GroupVersion.String() + "/namespaces/{namespace}/manifest").To(target))
+	return ws, nil
+}
+
+func alive(request *restful.Request, response *restful.Response) {
+	res := map[string]interface{}{"status": "ok"}
+	response.WriteEntity(res)
+}
+
+func (m manifest) mapManifest(_ context.Context, request interface{}) (interface{}, error) {
+	log := logging.DefaultLogger()
+
+	vm := request.(*endpoints.PutObject).Payload.(*v1.VM)
+
+	err := virt_manifest.AddMinimalVMSpec(vm)
+	if err != nil {
+		log.Error().Reason(err).Msg("Unable to map domain")
+		return nil, middleware.NewBadRequestError(err.Error())
+	}
+
+	connection, err := virtwrap.NewConnection(m.libvirtUri, m.libvirtUser, m.libvirtPasswd, 0)
+	if err != nil {
+		return nil, middleware.NewGatewayTimeoutError(err)
+	}
+
+	mappedVm, err := virt_manifest.MapVM(connection, vm)
+	if err != nil {
+		return nil, middleware.NewBadRequestError(err.Error())
```

Comment:
When looking at the MapVM errors, this is a classical 500. It either fails becuase our mapper is not ok (bug), or because marshalling our struct to xml  fails (again bug).